src/main/java/com/dmgrt/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.dmgrt.cinema.dao.impl;
+
+import com.dmgrt.cinema.dao.MovieSessionDao;
+import com.dmgrt.cinema.lib.Dao;
+import com.dmgrt.cinema.models.MovieSession;
+import com.dmgrt.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl extends AbstractDao<MovieSession> implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        LocalDateTime startOfDay = date.atStartOfDay();
+        LocalDateTime endOfDay = date.atTime(LocalTime.MAX);
```

Comment:
these variables are redundant here, just pass values to setParameter method right away